python/tests/test_wright_fisher.py

```diff
@@ -598,27 +598,43 @@ def test_simplify_tables(self, ts, nsamples):
     def test_simplify_keep_unary_in_individuals(self, ts, nsamples):
         np.random.seed(123)
         ts = tsutil.mark_metadata(ts, "nodes")
+        sub_samples = random.sample(list(ts.samples()), min(nsamples, ts.num_samples))
         random_nodes = np.random.choice(ts.num_nodes, ts.num_nodes // 2)
         ts = tsutil.insert_individuals(ts, random_nodes)
         ts = tsutil.mark_metadata(ts, "individuals")
-        md_map = {
-            ts.node(n).metadata: ts.individual(ts.node(n).individual).metadata
-            for n in random_nodes
-        }
-
-        ts = ts.simplify(keep_unary_in_individuals=True)
 
-        # check there are some unary nodes, and each has the original individual
-        num_nonsample_unary = 0
-        for tree in ts.trees():
-            for n in tree.nodes():
-                if tree.num_children(n) == 1:
-                    node = ts.node(n)
-                    if node.is_sample():
-                        continue
-                    num_nonsample_unary += 1
-                    assert node.individual >= 0
-                    assert node.metadata in md_map
-                    orig_metadata = md_map[node.metadata]
-                    assert orig_metadata == ts.individual(node.individual).metadata
-        assert num_nonsample_unary > 0
+        sts = ts.simplify(sub_samples, keep_unary_in_individuals=True)
+        # check samples match
+        assert sts.num_samples == len(sub_samples)
+        for n, sn in zip(sub_samples, sts.samples()):
+            print("::::", n, sn)
```

Comment:
do you want this print in there?

Reply:
whoops sorry

Reply:
NP. I do this *all* the time (just ask Jerome)